pubsub/redis/redis.go

```diff
@@ -183,11 +194,21 @@ func createRedisMessageWrapper(ctx context.Context, stream string, handler pubsu
 		}
 	}
 
+	var metadata map[string]string
+	if metadataValue, exists := msg.Values["metadata"]; exists && metadataValue != nil {
+		metadataBytes := metadataValue.(string)
+		err := json.Unmarshal([]byte(metadataBytes), &metadata)
+		if err != nil {
+			r.logger.Warnf("Could not extract metadata for Redis message %s: %v", msg.ID, err)
```

Comment:
This seems to be better than getting the message stuck because metadata could not be handled.